membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -105,11 +110,12 @@ class AccountController extends LazyLogging {
       freeOrPaidSub <- OptionEither(tp.subService.either[F, P](contact).map(_.leftMap(message => s"couldn't read sub from zuora for crmId ${contact.salesforceAccountId} due to $message")))
       details <- OptionEither.liftOption(tp.paymentService.paymentDetails(freeOrPaidSub).map(\/.right))
       sub = freeOrPaidSub.fold(identity, identity)
-      account <- OptionEither.liftOption(tp.zuoraService.getAccount(sub.accountId).map(\/.right).recover { case x => \/.left(s"error receiving account for subscription: ${sub.name} with account id ${sub.accountId}. Reason: $x") })
-      publicKey = account.paymentGateway.flatMap(tp.stripeServicesByPaymentGateway.get).map(_.publicKey)
+      account <- OptionEither.liftOption(tp.zuoraService.getAccount(sub.accountId).map(\/.right).recover { case x => \/.left(s"error retrieving account for subscription: ${sub.name} with account id ${sub.accountId}. Reason: $x") })
+      billToContact <- OptionEither.liftOption(tp.zuoraService.getContact(account.billToId).map(\/.right).recover { case x => \/.left(s"error retrieving bill to contact id (${account.billToId}) for account id ${account.id}. Reason $x")})
```

Comment:
Is this an extra network call? It's a shame that we can't just get what we need from the account...